website/pages/input-unions.mdx

```diff
@@ -0,0 +1,72 @@
+---
+title: Input Unions
+---
+
+import { Tabs } from 'nextra/components';
+
+Some inputs will behave differently depending on what input we choose. Let's look at the case for
+a field named `product`, we can fetch a `Product` by either its `id` or its `name`. Currently we'd
+make a tradeoff for this by introducing two arguments that are both nullable, now if both are passed
+as null we'd have to handle that in code. To fix this the `@oneOf` directive was introduced so we
+can create these input-unions without sacrificing the strictly typed nature of our GraphQL Schema.
+
+<Tabs items={['SDL', 'Code']}>
+    <Tabs.Tab>
+```js
+const schema = buildSchema(`
+  type Product {
+    id: ID!
+    name: String!
+  }
+
+  input ProductInput @oneOf {
+    id: ID
+    name: String
+  }
+
+  type Query {
+    product(input: ProductInput!): Product
+  }
```

Comment:
`ProductInput` is generally used for CRUD-style mutations. Let's use a different input object name here:

```suggestion
  input ProductSpecifier @oneOf {
    id: ID
    name: String
  }

  type Query {
    product(by: ProductSpecifier!): Product
  }
```

Reply:
Maybe we should expand this so it demonstrates usage with non-scalars too?

```suggestion
  input ProductLocation {
    aisleNumber: Int!
    shelfNumber: Int!
    positionOnShelf: Int!
  }
  
  input ProductSpecifier @oneOf {
    id: ID
    name: String
    location: ProductLocation
  }

  type Query {
    product(by: ProductSpecifier!): Product
  }
```